data/org.freedesktop.portal.Usb.xml

```diff
@@ -0,0 +1,222 @@
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library. If not, see <http://www.gnu.org/licenses/>.
+
+ Author: Georges Basile Stavracas Neto <georges.stavracas@gmail.com>
+         Ryan Gonzalez <rymg19+github@gmail.com>
+-->
+
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <!--
+      org.freedesktop.portal.Usb:
+      @short_description: Portal for USB device access
+
+      This interface lets sandboxed applications monitor and request
+      access to connected USB devices.
+
+      Applications should prefer specialized portals for specific
+      device types, such as the Camera portal for cameras.
+
+      This documentation describes version 1 of this interface.
+    -->
+  <interface name="org.freedesktop.portal.Usb">
+
+    <!--
+        CreateSession:
+        @options: Vardict with optional further information
+        @handle: Object path for the :ref:`org.freedesktop.portal.Request` object representing this call
+
+        Creates an USB monitoring session.
+
+        Supported keys in the @options vardict include:
+
+        * ``session_handle_token`` (``s``)
+
+          A string that will be used as the last element of the session handle. Must be a valid
+          object path element. See the #org.freedesktop.portal.Session documentation for
+          more information about the session handle.
+    -->
+    <method name="CreateSession">
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="session_handle" direction="out"/>
+    </method>
+
+    <!--
+        EnumerateDevices:
+        @options: Vardict with optional further information
+
+        Enumerates all connected USB devices that this application has permission
+        to see.
+
+        The following results are returned in the @device vardict:
+
+        * ``id`` (``s``)
+
+          The portal-specific unique identifier of the device.
+
+        * ``parent`` (``s``)
+
+          Device ID of the parent device.
+
+        * ``readable`` (``b``)
+
+          Whether the device can be opened for reading with
+          org.freedesktop.portal.Usb.AcquireDevices(). If not present, then
+          it should be assumed to be false.
+
+        * ``writable`` (``b``)
+
+          Whether the device can be opened for writing with
+          org.freedesktop.portal.Usb.AcquireDevices(). If not present, then
+          it should be assumed to be false.
+
+        * ``device-file`` (``s``)
+
+          A string path to the device node inside the /dev filesystem.
+
+        * ``properties`` (``a{sv}``)
+
+          A list of udev properties that this device has. These properties
```

Comment:
I'm really worried about this. It essentially makes udev a stable API even though it really is not.